node/bft/storage-service/src/persistent.rs

```diff
@@ -152,66 +179,84 @@ impl<N: Network> StorageService<N> for BFTPersistentStorage<N> {
         aborted_transmission_ids: HashSet<TransmissionID<N>>,
         mut missing_transmissions: HashMap<TransmissionID<N>, Transmission<N>>,
     ) {
-        // Inserts the following:
-        //   - Inserts **only the missing** transmissions from storage.
-        //   - Inserts the certificate ID into the corresponding set for **all** transmissions.
-        'outer: for transmission_id in transmission_ids {
-            // Retrieve the transmission entry.
+        // First, handle the non-aborted transmissions.
+        for transmission_id in transmission_ids {
+            // Try to fetch from the persistent storage.
             match self.transmissions.get_confirmed(&transmission_id) {
                 Ok(Some(entry)) => {
+                    // The transmission exists in storage; update its certificate IDs.
                     let (transmission, mut certificate_ids) = cow_to_cloned!(entry);
-                    // Insert the certificate ID into the set.
                     certificate_ids.insert(certificate_id);
-                    // Update the transmission entry.
-                    if let Err(e) = self.transmissions.insert(transmission_id, (transmission, certificate_ids)) {
+
+                    // Update the persistent storage.
+                    if let Err(e) =
+                        self.transmissions.insert(transmission_id, (transmission.clone(), certificate_ids.clone()))
+                    {
                         error!("Failed to insert transmission {transmission_id} into storage - {e}");
-                        continue 'outer;
                     }
+
+                    // Also, update the cache.
+                    self.cache_transmissions.write().put(transmission_id, (transmission, certificate_ids));
                 }
                 Ok(None) => {
-                    // Retrieve the missing transmission.
-                    let Some(transmission) = missing_transmissions.remove(&transmission_id) else {
-                        if !aborted_transmission_ids.contains(&transmission_id)
-                            && !self.contains_transmission(transmission_id)
```

Comment:
note to reviewers: apparently this call (`contains_transmission`) was redundant (as it checks `self.transmissions` and `self.aborted_transmissions_id`, which would have already been done by now)

Reply:
I'm investigating a validator halting on a devnet running with this PR, made me look deeper into this. I don't think `contains_transmission` is redundant, as it additionally check `self.aborted_transmissions`. Without doing a full analysis, @kpandl can you revert to the original logic? 

```
let Some(transmission) = missing_transmissions.remove(&transmission_id) else {
                        if !aborted_transmission_ids.contains(&transmission_id)
                            && !self.contains_transmission(transmission_id)
```